src/daft-connect/src/spark_analyzer.rs

```diff
@@ -294,22 +293,117 @@ impl SparkAnalyzer<'_> {
             debug!("Ignoring options: {options:?}; not yet implemented");
         }
 
-        if !predicates.is_empty() {
-            debug!("Ignoring predicates: {predicates:?}; not yet implemented");
+        // TODO: allow user to set handling for unused options. either ignore or error.
+        fn check_unused_options(
+            format: &str,
+            options: &HashMap<String, String>,
+        ) -> ConnectResult<()> {
+            let mut unimplemented_options = Vec::new();
+            for (key, value) in options {
+                warn!("Option '{key}' is not yet implemented");
+                unimplemented_options.push(format!(r#""{}"={}"#, key, value));
+            }
+            if unimplemented_options.is_empty() {
+                Ok(())
+            } else {
+                Err(ConnectError::not_yet_implemented(format!(
+                    "[{unimplemented_options}] options for {format}",
+                    unimplemented_options = unimplemented_options.join(", ")
+                )))
+            }
```

Comment:
Could you please update this to only include the keys? I believe there's ambiguity with having both the key and value. For example, is `headers=0` not supported because of "headers" or because "0"?

I think you've already covered the latter, so perhaps just using option keys here. If the value is something not supported, we can have more precise error messages while parsing that option's value.